### DISCUSSION TITLE
Help with Automations - Multiple actions in Loop

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#7753] [Help] [answered]
I am trying to achieve the following scenario and i don't think i figured out exactly how automatio loop and action chaining work.

Scenario (MySQL)
I have a table, called sms_queue, with id, batch_id, phone, message, status.
I fill this table with outgoing sms messages
I want to create an automation, that will poll that table, select all messages with status "queued" and try to deliver them using an external API call.
Lets assume this Automation is triggered by an App Action on the press of a button.

The process flow would be :
- steps.1 - Query Rows
- LOOP > steps.1.rows
- steps.2 : External Data Connector SMS GW to {{ loop.currentItem.phone }} 
- steps.3 : External Data Connector Query that updates row setting status to {{ steps.2.success }} and take row_id as input parameter {{ loop.currentItem.id }}
- steps.4 : Delay 1000ms
- END LOOP
  
But how do I set multiple steps inside a loop ? 

I must be missing something really simple here !

**Comment by melohagan** [comment, marked as answer]
Hey @poirazis 

The looping applies on a per step basis. So I think unfortunately what you'd need to do is:

 - LOOP (steps.1.rows) > EDC send SMS
 - LOOP (steps.1.rows) > EDC update the rows status'

Instead of doing looping, it may be better to try using a **Row Created** trigger on a "queue" table.
Or maybe some sort of webhook that is triggered anytime a SMS needs to be sent.

**Reply by poirazis** [reply]
Hey @melohagan 
I though about the table / on Row Created Trigger but what would happen if i Insert say, 5000 rows rapidly ?
There would be 5000 Triggered Automations launched ?

**Reply by melohagan** [reply]
In the Budibase backend automations are queued, so it'll only execute one at a time.

**Reply by poirazis** [reply]
Thank you @melohagan !
I will rewire my logic then!
